server/src/unifycr_cmd_handler.c

```diff
@@ -423,6 +423,61 @@ static void unifycr_mount_rpc(hg_handle_t handle)
 }
 DEFINE_MARGO_RPC_HANDLER(unifycr_mount_rpc)
 
+static void unifycr_unmount_rpc(hg_handle_t handle)
+{
+    /* get input params */
+    unifycr_unmount_in_t in;
+    int ret = HG_Get_input(handle, &in);
+    assert(ret == HG_SUCCESS);
+
+    /* read app_id and client_id from input */
+    int app_id    = in.app_id;
+    int client_id = in.local_rank_idx;
+
+    /* lookup app_config for given app_id */
+    app_config_t *app_config =
+        (app_config_t *) arraylist_get(app_config_list, app_id);
+
+    /* TODO: do cleanup here */
+
+    /* get thread id for this client */
+    int thrd_id = app_config->thrd_idxs[client_id];
+
+    /* look up thread control structure */
+    thrd_ctrl_t *thrd_ctrl = (thrd_ctrl_t *)arraylist_get(thrd_list, thrd_id);
+
+    /* shutdown the delegator thread */
+    rm_cmd_exit(thrd_ctrl);
+
+    /* detach from the request shared memory */
+    unifycr_shm_free(app_config->req_buf_name[client_id],
+                     app_config->req_buf_sz,
+                     &(app_config->shm_req_bufs[client_id]));
+    app_config->shm_req_bufs[client_id] = NULL;
+
+    /* detach from the read shared memory buffer */
+    unifycr_shm_free(app_config->recv_buf_name[client_id],
+                     app_config->recv_buf_sz,
+                     &(app_config->shm_recv_bufs[client_id]));
+    app_config->shm_recv_bufs[client_id] = NULL;
+
+    /* destroy the sockets except for the ones for acks */
+    sock_sanitize_cli(client_id);
+
+    /* build output tructure to return to caller */
+    unifycr_mount_out_t out;
+    out.ret = ret;
```

Comment:
Looks like ret is last set from the HG_Get_input call here.  I think we should return something like UNIFYCR_SUCCESS / FAILURE instead.  For now, we could just have it always return SUCCESS, and then add logic later to conditionally return FAILURE or error code if the unmount operation failed for some reason.

If other rpc handlers are similar, let's create an issue and clean this up in another PR.

Reply:
@adammoody Makes sense. I updated to return UNIFYCR_SUCCESS.